activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -465,6 +465,10 @@ def to_r
       utc.to_r
     end
 
+    # Ruby protocol for +Time.at+ requires +to_int+ to be defined.
+    def to_int
+    end
```

Comment:
Is this method really supposed to be blank? I figured it'd be an alias for `to_i`, and return a number.

Reply:
I can't think of a scenario where its return value is used.

Please correct me if I'm mistaken, there's this convention for `to_` methods, there are e.g. `to_s` and `to_str` (the same applies to `to_i`/`to_int`, `to_h`/`to_hash`). The latter works if the object is of a target class. So, a string-like object would respond to `to_str`, and that would be used e.g. in `"a" + object`, a `to_str` would be called on it. This is to avoid the implicit coercion. So, the short versions are for explicit coercion and the longer for implicit.

The same, I believe applies in this case. I don't want to make `ActiveSupport::TimeWithZone` implicitly coercible to an integer, that would make `1 + Time.current` possible, but that doesn't make any sense. I rather prefer it to return `nil` and fail early.

Reply:
Then perhaps the method should return `nil` as opposed to being empty.  And add some comments explaining why since not everyone is going to read this PR.

Reply:
Is there a test for this? I'm not seeing one. If not, there definitely should be.

Reply:
The test that covers the new behaviour is:
```ruby
- assert_raise(TypeError) { assert_equal(Time.utc(2000, 1, 1, 0, 0, 0), Time.at(ActiveSupport::TimeWithZone.new(Time.utc(2000, 1, 1, 0, 0, 0), ActiveSupport::TimeZone["UTC"]), 0)) }
+ time_with_zone = ActiveSupport::TimeWithZone.new(time_with_nsec, ActiveSupport::TimeZone["UTC"])
+ assert_equal time_with_zone, Time.at(time_with_zone)
```

Reply:
> should return `nil` as opposed to being empty

This may lead to confusion since `nil` is a specific value, but actually it can equally be `"hello"`, or `Float::Infinity`, or even `raise "ooops"`. This method is never called, so its return value or its behaviour doesn't really matter.

> add some comments explaining why

Sure, that makes total sense.

Reply:
There is a comment actually:
```ruby
# Ruby protocol for +Time.at+ requires +to_int+ to be defined.
```

Would it be sufficient to add something like?
```ruby
# In case `to_int` is not defined, the `Time.at` performs a lossy conversion that is causing the loss of nanosecond precision.
```

Reply:
Added even more details in the comment. Please take another look.